tests/formparams/form_params_test.go

```diff
@@ -0,0 +1,398 @@
+			t.Errorf(`req.Form["fruit"] = %v, want { "apple" }`, req.Form["apple"])
+		}
+	})
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(status200OK)) {
+		t.Errorf("response status: got %s, want %s", string(resp), status200OK)
+	}
+}
+
+// Test whether passing a POST request with body and without Content-Length
+// yields a 400 Bad Request
+func TestFormParametersMissingContentLength(t *testing.T) {
+	postReq := []byte("POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" +
+		"veggie=potato\r\n" +
+		"\r\n")
+	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
+		t.Error("expected handler not to be called.")
+	})
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(status400BadReq)) {
+		t.Errorf("response status: got %s, want %s", string(resp), status400BadReq)
+	}
+}
+
+// Ensure passing a negative Content-Length or one that overflows integers results in a
+// 400 Bad Request
+func TestFormParametersBadContentLength(t *testing.T) {
+	tests := []struct {
+		name string
+		req  []byte
+	}{
+		{
+			name: "Negative Content-Length",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" +
+				"Content-Length: -1\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+		},
+		{
+			name: "Integer Overflow Content-Length",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" +
+				"Content-Length: 10000000000000000000000000\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			resp, err := requesttesting.MakeRequest(context.Background(), test.req, func(req *http.Request) {
+				t.Error("expected handler not to be called.")
+			})
+			if err != nil {
+				t.Fatalf("MakeRequest(): got err %v, want nil", err)
+			}
+			if !bytes.HasPrefix(resp, []byte(status400BadReq)) {
+				t.Errorf("response status: got %s, want %s", string(resp), status400BadReq)
+			}
+		})
+	}
+}
+
+// Tests behaviour when multiple Content-Length values are passed using three
+// testcases:
+// a) passing two Content-Length headers containing the same value
+// b) passing two Content-Length headers containing different values
+// c) passing one Content-Length header containing a list of equal values
+// For a) and c), RFC 7320, Section 3.3.2 says that the server can either accept
+// the request and use the duplicated value or reject it and for b) it should be
+// rejected. It is expected that Go will handle a) and c) in a similar manner
+func TestFormParametersDuplicateContentLength(t *testing.T) {
```

Comment:
In this case, I think it'd be better to split out the two failing ones into a separate `TestFormParametersDuplicateContentLengthFailing` test fixture.

The way it is right now it's very easy to miss the `t.Skip()`, and actually, it's incorrect in this case as you are calling it on the whole test parameter, instead of individual subtests. 

The approach I was suggesting earlier basically required having a separate field in the `tests` struct which would indicate whether this is to be skipped or not, but on a second though that would be unnecessarily complex.

Reply:
In [9a7e610](url), I separated the request that succeeds with duplicate Content-Length and the ones that dont. Also, I switched to tabletests